From cc220fc5dfe586c367265d5552327d555a80d413 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:40:47 -0400
Subject: [alsa] hda: Set internal mic as default input source on Dell XPS 13 9333

Message-id: <140972823573-416-patchset-jkysela@redhat.com>
Patchwork-id: 91112
O-Subject: [RHEL 7.1 PATCH 415/442] ALSA: hda - Set internal mic as default input source on Dell XPS 13 9333
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

If the laptop is powered on with a jack plugged in, independently on what
is plugged, the jack is treated as a microphone jack.

Initialize the capture source so that by default jacks are treated as
headphones jacks. This will also prevent pop noises on boot in case
headphones are plugged in since setting/unsetting mic-in as input source
causes a pop noise.

Bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=76611
Signed-off-by: Gabriele Mazzotta <gabriele.mzt@gmail.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: f38663ab5cc2d1bf359c0fd2b6b84a9f000e4388)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 69f2744..3ebb761 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -4103,8 +4103,19 @@ static void alc_fixup_dell_xps13(struct hda_codec *codec,
 {
  if (action == HDA_FIXUP_ACT_PROBE) {
   struct alc_spec *spec = codec->spec;
+  struct hda_input_mux *imux = &spec->gen.input_mux;
+  int i;
+
   spec->shutup = alc_no_shutup;
   codec->power_filter = alc_power_filter_xps13;
+
+  /* Make the internal mic the default input source. */
+  for (i = 0; i < imux->num_items; i++) {
+   if (spec->gen.imux_pins[i] == 0x12) {
+    spec->gen.cur_mux[0] = i;
+    break;
+   }
+  }
  }
 }
 
-- 
1.7.1